File: Linear Keyboard.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    char a[26];
    string l;
    int t;
    cin>>t;
    while(t){
        for(int i=0;i<26;i++)
            cin>>a[i];
        cin>>l;
        int k1,k2;
        int lans,ans;
        ans=0;
        for(int i=1;i<l.length();i++){
            int x=0;
            while(a[x]!=l[i-1])
                x++;
            k1=x;
            x=0;
            while(a[x]!=l[i])
                x++;
            k2=x;
            lans=abs(k1-k2);
            ans+=lans;
        }
        cout<<ans<<endl;
        t--;
    }
    return 0;
}

File: Linear Keyboard.py
k = int(input())
for i in range(k):
	keys = {c:i for i, c in enumerate(input())}
	word = input()
	ans = 0
	for i in range(1, len(word)):
		ans += abs(keys[word[i-1]] - keys[word[i]])
	print(ans)